### PR TITLE
Updating ExampleGen user guide with Date spec, Version spec, Width modifiers, and RangeConfig

### PR DESCRIPTION
**Pull request by 18jeffreyma**
Recently the following PRs were approved (or are currently under review):

- Version Spec ([PR#2138](https://github.com/tensorflow/tfx/pull/2138))
- Date Spec ([PR#2285](https://github.com/tensorflow/tfx/pull/2285))
- Span and Version spec width modifiers (approved and merged, [PR#2304](https://github.com/tensorflow/tfx/pull/2304))
- Support for Range Config in FileBasedExampleGen (under review, [PR#2339](https://github.com/tensorflow/tfx/pull/2339))

This CL updates the `examplegen.md` user guide for the changes and new features provided in these PRs. Some notable ones:
- Version spec is now supported.
- Date spec is now supported in adddition to Span spec.
- RangeConfig with width modifiers now allows for processing of specific spans, even with zero padding.

### CODE REVIEW

**Comment by 18jeffreyma** [comment]
@1025KB

File: docs/guide/examplegen.md [hunk 1]
```diff
@@ -215,13 +220,215 @@ examples = csv_input('/tmp')
...
 ```
 
-Note: Retrieving a certain span is not supported yet. You can only fix the
-pattern for now (for example, 'span-2/eval/*' instead of 'span-{SPAN}/eval/*'),
-but by doing this, span number stored in metadata will be zero.
+Retrieving a certain span can be done with RangeConfig, which is detailed below.
+
+### Date
+
+Note: this feature is only availible after TFX 0.22.1.
```
Comment (hughmiao) [inline docs/guide/examplegen.md:227]: s/0.22.1/0.23.0 ?

File: docs/guide/examplegen.md [hunk 2]
```diff
@@ -215,13 +220,215 @@ examples = csv_input('/tmp')
...
+                                            pattern='{YYYY}-{MM}-{DD}/eval/*')
+            ])
+examples = csv_input('/tmp')
+example_gen = CsvExampleGen(input=examples, input_config=input)
+```
 
 ### Version
 
-Note: Version is not supported yet
+Note: this feature is only availible after TFX 0.22.1.
```
Comment (hughmiao) [inline docs/guide/examplegen.md:290]: s/0.22.1/0.23.0 ?

File: docs/guide/examplegen.md [hunk 3]
```diff
@@ -215,13 +220,215 @@ examples = csv_input('/tmp')
...
+                example_gen_pb2.Input.Split(name='eval',
+                                            pattern='span-{SPAN}/ver-{VERSION}/eval/*')
+            ])
+examples = csv_input('/tmp')
+example_gen = CsvExampleGen(input=examples, input_config=input)
+```
+
+### Range Config
+
+Note: this feature is only available after TFX 0.22.1.
```
Comment (hughmiao) [inline docs/guide/examplegen.md:355]: ditto

File: docs/guide/examplegen.md [hunk 4]
```diff
@@ -215,13 +220,215 @@ examples = csv_input('/tmp')
...
+# required so TFX can correctly substitute spec with zero-padded span number.
+input = example_gen_pb2.Input(splits=[
+                example_gen_pb2.Input.Split(name='train',
+                                            pattern='span-{SPAN:2}/train/*'),
+                example_gen_pb2.Input.Split(name='eval',
+                                            pattern='span-{SPAN:2}/eval/*')
+            ])
+# Specify the span number to be processed here using StaticRange.
+# Note that ExampleGen only supports single-span static ranges. Thus,
+# for StaticRange, start_span_number must equal end_span_number.
```
Comment (hughmiao) [inline docs/guide/examplegen.md:385]: let's move this current constraint upfront before the example code.

**Comment by 18jeffreyma** [comment]
addressed latest round of feedback, thanks!

**Review by rcrowe-google — APPROVED** [review]
Sorry for the delay, LGTM

**Review by 1025KB — APPROVED** [review]

**Pull request merged**